core/include/detray/tools/constrained_step.hpp

```diff
@@ -0,0 +1,117 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// detray definitions
+#include <climits>
+#include <type_traits>
+
+#include "detray/definitions/qualifiers.hpp"
+
+namespace detray {
+
+// Add constraints for steppers
+namespace step {
+
+/// Direction in which the integration is performed
+enum direction : int {
```

Comment:
Let's call it `step_direction` as we also need to add `navigation_direction` in the future

Reply:
This is the navigation direction used in the stepper. It lives in the ```step``` namespace so that it should be no problem to add a ```direction``` ```enum``` in the ```navigator```, ```propagator``` etc

Reply:
I haven't checked step namespace right upthere